### PR TITLE
Setting enough cache_size_limit for torch.compile warmup

### PR DESCRIPTION
**Pull request by zehao-intel**
Fix the issue that warmup sometimes doesn't work because the default cache_size_limit is only 8 .

### CODE REVIEW

File: vllm/worker/habana_model_runner.py [hunk 1]
```diff
@@ -596,6 +596,11 @@ def _setup_buckets(self) -> None:
                 bucket for bucket in self.decode_buckets
                 if self._is_valid_bucket(bucket)
             ]
+        if not htorch.utils.internal.is_lazy() and not self.enforce_eager:
+            cache_size_limit = len(self.prompt_buckets)+len(self.decode_buckets)
```
Comment (anko-intel) [inline vllm/worker/habana_model_runner.py:600]:
```suggestion
            cache_size_limit = len(self.prompt_buckets) + len(self.decode_buckets)
```

File: vllm/worker/habana_model_runner.py [hunk 2]
```diff
@@ -596,6 +596,11 @@ def _setup_buckets(self) -> None:
...
+            torch._dynamo.config.cache_size_limit = cache_size_limit
+            torch._dynamo.config.accumulated_cache_size_limit = cache_size_limit*8
```
Comment (anko-intel) [inline vllm/worker/habana_model_runner.py:602]:
```suggestion
            torch._dynamo.config.accumulated_cache_size_limit = cache_size_limit * 8
```

**Review by anko-intel — APPROVED** [review]

File: vllm/worker/habana_model_runner.py [hunk 3]
```diff
@@ -1467,6 +1467,11 @@ def warmup_model(self, kv_caches: List[torch.Tensor]) -> None:
                     len(self.decode_buckets),
                     list(sorted(self.decode_buckets)))
 
+        if not htorch.utils.internal.is_lazy() and not self.enforce_eager:
+            cache_size_limit = len(self.prompt_buckets) + len(self.decode_buckets)
+            torch._dynamo.config.cache_size_limit = cache_size_limit
+            torch._dynamo.config.accumulated_cache_size_limit = cache_size_limit * 8
```
Comment (anko-intel) [inline vllm/worker/habana_model_runner.py:1473]: What is the origin of this "8"? 
Please add comment or calculate it from some other parameters

File: vllm/worker/habana_model_runner.py [hunk 4]
```diff
@@ -1467,6 +1467,11 @@ def warmup_model(self, kv_caches: List[torch.Tensor]) -> None:
                     len(self.decode_buckets),
                     list(sorted(self.decode_buckets)))
 
+        if not htorch.utils.internal.is_lazy() and not self.enforce_eager:
+            cache_size_limit = len(self.prompt_buckets) + len(self.decode_buckets)
+            torch._dynamo.config.cache_size_limit = cache_size_limit
```
Comment (anko-intel) [inline vllm/worker/habana_model_runner.py:1472]: maybe we have to check if user already do not setup greater values then we are setting. Some think like  limit = max (limit, new_value)

File: vllm/worker/habana_model_runner.py (hunk 3 above)
Comment (zehao-intel) [reply vllm/worker/habana_model_runner.py:1473]: I just followed the original ratio between the 2 parameters(8 for cache_size_limit and 64 for accumulated_cache_size_limit).

File: vllm/worker/habana_model_runner.py (hunk 4 above)
Comment (zehao-intel) [reply vllm/worker/habana_model_runner.py:1472]: Thanks. I think this should be more reasonable. I will modify the code.

**Review by anko-intel — APPROVED** [review]

**Comment by zehao-intel** [comment]
> please fix formatting (format.sh script) so that static checks pass, otherwise lgtm

Thanks for your comments, the format has been fixed now.
